File: day/d3/part1.py
# read file
# declare bool list
# sum = 0
# for each line
# size/2, 

def get_priority(n: int) -> int:
    return n-97 if n>97 else n-39

def main():
    first_department = [False]*52
    file = open('input.txt', 'r')
    Lines = file.readlines()
    sum = 0
    for line in Lines:
        half = len(line)//2
        for i in range(0, half):
            ascii = get_priority(ord(line[i]))
            if not first_department[ascii]:
                first_department[ascii] = True
        for i in range(half, len(line)):
            ascii = get_priority(ord(line[i]))
            if first_department[ascii]:
                sum += (ascii + 1)
                break
        first_department = [0 for _ in first_department]
    print(sum)

if __name__=="__main__":
    main()

File: day/d4/part1.c
#include <stdio.h>

struct Interval {
    int begin, end;
};

int contains(struct Interval *i1,struct Interval *i2){
    return ((i1->begin <= i2->begin) && (i2->end <= i1->end)) || ((i2->begin <= i1->begin) && (i1->end <= i2->end));
}

int main() {
    // sum = 0, open file, read line,
    int sum = 0;
    char line[12];
    struct Interval i1, i2;
    FILE *f = fopen("input.txt", "r");
    while((fgets(line, 121, f)) != NULL) {
        sscanf(line, "%d-%d,%d-%d",&i1.begin, &i1.end, &i2.begin, &i2.end);
        if(contains(&i1, &i2)) {
            sum++;
        }
    }
    fclose(f);
    printf("Total: %d\n", sum);
    return 0;
}
